paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/base/BasicArrayResourceItem.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.paparazzi.internal.resources.base
+
+import app.cash.paparazzi.internal.resources.ResourceSourceFile
+import com.android.ide.common.rendering.api.ArrayResourceValue
+import com.android.resources.ResourceType
+import com.android.resources.ResourceVisibility
+
+/**
+ * Ported from: [BasicArrayResourceItem.java](https://cs.android.com/android-studio/platform/tools/base/+/18047faf69512736b8ddb1f6a6785f58d47c893f:resource-repository/main/java/com/android/resources/base/BasicArrayResourceItem.java)
+ *
+ * Resource item representing an array resource.
+ */
+class BasicArrayResourceItem(
+  name: String,
+  sourceFile: ResourceSourceFile,
+  visibility: ResourceVisibility,
+  private val elements: List<String>,
+  private val defaultIndex: Int
+) : BasicValueResourceItemBase(ResourceType.ARRAY, name, sourceFile, visibility), ArrayResourceValue {
+
+  init {
+    assert(elements.isEmpty() || defaultIndex < elements.size)
+  }
+
+  override fun getElementCount() = elements.count()
+
+  override fun getElement(index: Int) = elements[index]
+
+  override fun iterator(): MutableIterator<String> = elements.toMutableList().iterator()
+
```

Comment:
@jrodbx The `iterator()` from interface `java.lang.Iterable` is returning `java.util.Iterator`, so the return type would be MutableIterator after converting to Kotlin. Is there any good way to convert it to immutable Iterator in Kotlin?

Reply:
Doesn't look like that's possible, due to Java interop:

https://stackoverflow.com/a/61917100
https://kotlinlang.org/docs/java-interop.html#notation-for-platform-types

Reply:
Tip: maybe use `= java.util.Collections.unmodifiableList(elements).iterator()`, it still returns `MutableIterator`, but it'll catch if anyone is trying to modify the returned object as soon as it happens, because all the modifications would just go to a black-hole copied collection otherwise, and potential issues would be either hidden or come out in a totally unrelated part of the code.

Reply:
Alternatively this would work:
```kotlin
	@Suppress("RETURN_TYPE_MISMATCH_ON_OVERRIDE") // This is suppressing a compile error https://github.com/JetBrains/kotlin/blob/master/compiler/frontend/src/org/jetbrains/kotlin/diagnostics/rendering/DefaultErrorMessages.java#L950
	override fun iterator(): Iterator<String> = elements.iterator()
```
because it'll compile to the right bytecode:
```java
   @NotNull
   public Iterator iterator() {
      return this.elements.iterator();
   }
```
but this makes the object actually mutable from Java.

Reply:
> Tip: maybe use `= java.util.Collections.unmodifiableList(elements).iterator()`, it still returns `MutableIterator`, but it'll catch if anyone is trying to modify the returned object as soon as it happens, because all the modifications would just go to a black-hole copied collection otherwise, and potential issues would be either hidden or come out in a totally unrelated part of the code. Additionally it won't create an extra copy array, but I guess this would be negligible anyway for Android Array Resources.

@TWiStErRob this solution looks good, will update the PR

Reply:
> Tip: maybe use `= java.util.Collections.unmodifiableList(elements).iterator()`, it still returns `MutableIterator`, but it'll catch if anyone is trying to modify the returned object as soon as it happens, because all the modifications would just go to a black-hole copied collection otherwise, and potential issues would be either hidden or come out in a totally unrelated part of the code. Additionally it won't create an extra copy array, but I guess this would be negligible anyway for Android Array Resources.

@TWiStErRob this solution looks good, will update the PR